bpf/lib/ipv4.h

```diff
@@ -143,20 +138,32 @@ ipv4_handle_fragment(struct __ctx_buff *ctx,
 		.pad = 0,
 	};
 
-	update_metrics(ctx_full_len(ctx), dir, REASON_FRAG_PACKET);
+	is_fragment = ipv4_is_fragment(ip4);
 
-	not_first_fragment = ipv4_is_not_first_fragment(ip4);
-	if (has_l4_header)
-		*has_l4_header = !not_first_fragment;
+	if (unlikely(is_fragment)) {
+		update_metrics(ctx_full_len(ctx), dir, REASON_FRAG_PACKET);
 
-	if (likely(not_first_fragment))
-		return ipv4_frag_get_l4ports(&frag_id, ports);
+		not_first_fragment = ipv4_is_not_first_fragment(ip4);
+		if (has_l4_header)
+			*has_l4_header = !not_first_fragment;
 
-	/* First logical fragment for this datagram (not necessarily the first
-	 * we receive). Fragment has L4 header, we can retrieve L4 ports and
-	 * create an entry in datagrams map.
-	 */
-	return ipv4_frag_register_datagram(ctx, l4_off, dir, &frag_id, ports);
+		if (likely(not_first_fragment))
+			return ipv4_frag_get_l4ports(&frag_id, ports);
+	}
+
+	ret = ctx_load_bytes(ctx, l4_off, ports, 4);
```

Comment:
Can we keep the `/* load sport + dport into tuple */` comment? It's not super obvious that we are loading both on a quick read.